lib/matplotlib/backends/backend_qt5agg.py

```diff
@@ -71,7 +71,11 @@ def paintEvent(self, e):
         In Qt, all drawing should be done inside of here when a widget is
         shown onscreen.
         """
-        FigureCanvasAgg.draw(self)
+        if not hasattr(self, "_agg_redraw_flag"):
+            self._agg_redraw_flag = True
+        if self._agg_redraw_flag:
+            self._agg_redraw_flag = False
```

Comment:
please flip the order on these lines so it is only cleared after the Agg draw has succeeded with out exception.